### PR TITLE
Closes #12: spring-boot-devtools should not be used on production

### LINKED ISSUE #12: spring-boot-devtools should not be used on production
Spring Boot developer tools are automatically disabled when running a fully packaged application (eg. using `java -jar`).
This can be overridden but it is not recommended.

From https://docs.spring.io/spring-boot/docs/2.5.4/reference/html/using.html#using.devtools

> This must not be done in a production environment where running devtools is a security risk.

We use `FileSystemWatcher` from devtools and should find a replacement.

### PR DESCRIPTION
**Pull request by marekhorst**
Removing spring-boot-devtools dependency after reimplementing file syncing to rely directly on the WatchService.

### CODE REVIEW

File: metrics-delivery-endpoint-impl/pom.xml [hunk 1]
```diff
@@ -30,11 +30,6 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-web</artifactId>
         </dependency>
-        
-        <dependency>
-            <groupId>org.springframework.boot</groupId>
-            <artifactId>spring-boot-devtools</artifactId>
-        </dependency>
```
Comment (mpol) [inline metrics-delivery-endpoint-impl/pom.xml:37]: We might want to actually keep it for the automatic restarts after recompilation?
Just set it to optional [as the docs suggest](https://docs.spring.io/spring-boot/docs/2.5.4/reference/htmlsingle/#using.devtools).

Although I'm still not sure how useful it really is during development.

File: metrics-delivery-endpoint-impl/src/main/java/eu/openaire/mas/delivery/mapping/JsonFileBasedMappingProvider.java [hunk 2]
```diff
@@ -142,4 +160,53 @@ private static String getInputStreamContents(InputStream is) throws IOException
...
+                            } else if (event.kind().equals(StandardWatchEventKinds.ENTRY_CREATE)
+                                    || event.kind().equals(StandardWatchEventKinds.ENTRY_MODIFY)) {
+                                log.debug("handling file related event by reinitializing the mappings");
+                                initializeMappings();
+                            } else {
+                                log.warn("unsupported event kind: " + event.kind());
+                            }
+                            log.info("hanlding operation: " + event.kind() + " on file: " + event.context() + " is done");
+                        } else {
+                            log.info("got notification unrelated to the mapping file: " + event.context());
```
Comment (mpol) [inline metrics-delivery-endpoint-impl/src/main/java/eu/openaire/mas/delivery/mapping/JsonFileBasedMappingProvider.java:197]: These two could be debug, I think. Especially the second one.

File: metrics-delivery-endpoint-impl/src/main/java/eu/openaire/mas/delivery/mapping/JsonFileBasedMappingProvider.java [hunk 3]
```diff
@@ -142,4 +160,53 @@ private static String getInputStreamContents(InputStream is) throws IOException
...
+        @Override
+        public void run() {
+            try {
+                WatchKey key;
+                while ((key = watchService.take()) != null) {
+                    for (WatchEvent<?> event : key.pollEvents()) {
+
+                        if (isMappingFile((Path) event.context())) {
+                            if (event.kind().equals(StandardWatchEventKinds.ENTRY_DELETE)) {
+                                log.debug("handling file related event by removing all the mappings");
```
Comment (mpol) [inline metrics-delivery-endpoint-impl/src/main/java/eu/openaire/mas/delivery/mapping/JsonFileBasedMappingProvider.java:186]: And this one I would probably make info.

**Review by mpol — APPROVED** [review]
A couple of small comments about logging levels, but generally seems to work.

File: metrics-delivery-endpoint-impl/pom.xml (hunk 1 above)
Comment (marekhorst) [reply metrics-delivery-endpoint-impl/pom.xml:37]: OK, as discussed during the call, we ain't gonna need it ;)

File: metrics-delivery-endpoint-impl/src/main/java/eu/openaire/mas/delivery/mapping/JsonFileBasedMappingProvider.java (hunk 3 above)
Comment (marekhorst) [reply metrics-delivery-endpoint-impl/src/main/java/eu/openaire/mas/delivery/mapping/JsonFileBasedMappingProvider.java:186]: OK, made it an info and also changed ln 190 to keep both in line.

**Comment by marekhorst** [comment]
Closed with 48714a663214e6a785a4f8f20654e382d72358f3 after merge and squash.

**Pull request closed without merging**